giraffe/src/components/Gauge.tsx

```diff
@@ -171,9 +171,18 @@ const drawSegmentedGauge = (
   colors: Color[],
   gaugeSize: number
 ): void => {
-  const sortedColors = Array.isArray(colors)
-    ? colors.sort((color1, color2) => color1.value - color2.value)
-    : []
+  const thresholdColors = []
+  if (Array.isArray(colors)) {
+    colors.filter(color => {
+      if (color.value >= minValue && color.value <= maxValue) {
+        thresholdColors.push(color)
+      }
+    })
+  }
```

Comment:
this isn't quite how `filter` is supposed to work.

broadly, there are two types of iteration functions on [Array's prototype.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) There are functions that iterate over every item of an array like `forEach` and `for... of` and there are functions that iterate over every item of an array and return results, like `map` or `filter`. When to use them comes from intent.

`forEach` is used when you need to do modifications - a good example is looping over every item of an array and converting it into a property of an object.

`map` and `filter` are meant to take an array and make a different array from them. `map` and `filter` both return new arrays.

`filter` works by looping over each item of the array and applying the function that you supply to that item. if the item passes the function's check, it's included in the array. if it fails, it is not included. the function you pass to `filter` should return `true` or `false`, and not have any side effects. calling `push` on an array is a side effect - rather than `push`ing to an outside array, you'll want this function to instead return `true` if the item should be in `thresholdColors` and `false` if it shouldn't . you'll also want to set the value of `threshColors` to be the return value of `filter`:

```ts
let thresholdColors = []
if (Array.isArray(colors)) {
  thresholdColors = colors.filter(// you implement this function to return true or false
```